lib/child_process.js

```diff
@@ -504,7 +504,7 @@ function normalizeSpawnArguments(file, args, options) {
   var env = options.env || process.env;
   var envPairs = [];
 
-  for (const key of Object.keys(env)) {
+  for (var key in env) {
```

Comment:
I would add a comment that flag that states that copying the prototype values is intentional.

Reply:
I added a comment to the commit before landing.